src/Lock.php

```diff
@@ -88,31 +88,81 @@ public static function showForItem(CommonDBTM $item)
             echo "<th width='10'>";
             Html::showCheckbox(['criterion' => ['tag_for_massive' => 'select_' . $lockedfield::getType()]]);
             echo "</th>";
-            echo "<th>" . $lockedfield->getTypeName() . "</th></tr>";
+            echo "<th>" . $lockedfield->getTypeName() . "</th>";
+            echo "<th>" . __('Itemtype') . "</th>";
+            echo "<th>" . __('Last inventoried value')  . "</th></tr>";
 
-            //inventory locked fields
-            $locked_iterator = $DB->request([
+            $subquery = [];
+
+            //get locked field for current itemtype
+            $subquery[] = new \QuerySubQuery([
+                'SELECT' => $lockedfield->getTable() . ".*",
                 'FROM'   => $lockedfield->getTable(),
                 'WHERE'  => [
                     'OR' => [
                         [
-                            'itemtype'  => $itemtype,
-                            'items_id'  => $ID
+                            $lockedfield->getTable() . '.itemtype'  => $itemtype,
+                            $lockedfield->getTable() . '.items_id'  => $ID
                         ], [
-                            'itemtype'  => $itemtype,
-                            'is_global' => 1
+                            $lockedfield->getTable() . '.itemtype'  => $itemtype,
+                            $lockedfield->getTable() . '.is_global' => 1
                         ]
                     ]
                 ]
             ]);
 
+            //get other locked related to curren itemtype
+            //where GLPI allows to update their data
+            $inventory_lockable_objects = [
```

Comment:
I do not understand why you limit to those 2 itemtypes only? It should be all of `inventory_lockable_objects` as far as I know, isn't it?

Reply:
Agree with that 
inventory_lockable_objects need to be review, some objects are not in this list

Reply:
Yes. Keep in mind `inventory_lockable_objects` has been designed for global locks, the dynamic lock mechanism can be done on any field of any object.